File: src/lib/software/paintdotnet.py
from PySide6.QtCore import Slot
from src.lib.software import BaseSoftware, SoftwareCategory
from src.lib.github_release_scraper import GithubReleaseScraper

class PaintDotNet(BaseSoftware):
    def __init__(self):
        super().__init__()

        self._gh = GithubReleaseScraper('paintdotnet', 'release', self)
        self._gh.releases_scraped.connect(self._on_releases_scraped)

        self.key = 'paint-dot-net'
        self.name = 'Paint.NET'
        self.category = SoftwareCategory.Creative
        self.download_name = 'Paint.NET.zip'
        self.is_archive = True
        self.should_cache_url = True
        self.requires_admin = False
        self.icon = 'paintdotnet.png'
        self.homepage = 'https://getpaint.net'

    @Slot(list)
    def _on_releases_scraped(self, releases: list[str]):
        asset = next((release for release in releases if release.endswith('.install.x64.zip')), None)
        if asset:
            self.url_resolved.emit(asset)
        else:
            self.url_resolve_error.emit(self.ResolveError.GitHubAssetNotFoundError)

    def resolve_download_url(self):
        self._gh.get_repo_releases()

File: packages/app/src/services/software/definitions/paintdotnet.ts
import { ok, err } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import { GithubReleasesScraper } from '../githubReleasesScraper';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'paint-dot-net';
	public name = 'Paint.NET';
	public category = [SoftwareCategory.Creative];
	public downloadName = 'Paint.NET.zip';
	public isArchive = true;
	public shouldCacheUrl = true;
	public requiresAdmin = false;
	public deprecated = false;
	public icon = 'paintdotnet.png';
	public homepage = 'https://getpaint.net';

	public async resolveDownloadUrl() {
		const owner   = 'paintdotnet';
		const repo    = 'release';
		const release = await GithubReleasesScraper.getRelease(owner, repo, r => r.endsWith('.install.x64.zip'));
		if (release.isErr()) {
			return err(release.error);
		}

		return ok(release.value);
	}
}
